app/js/enhancedDate.js

```diff
@@ -1,5 +1,99 @@
-/* 
+/*
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ */
 
+var enhancedDate = (function iffe() {
+  'use strict';
+  function now() {
+    return new Date();
+  }
+
+  var dateObjects = {
+    dateTime: now(),
+    monthNames: ['January',
+      'February',
+      'March',
+      'April',
+      'May',
+      'June',
+      'July',
+      'August',
+      'September',
+      'October',
+      'November',
+      'December'],
+    dayNames: ['Sunday',
+      'Monday',
+      'Tuesday',
+      'Wednesday',
+      'Thursday',
+      'Friday',
+      'Saturday']
+  };
+
+  return {
+    setDate: function setDate(e) {
+      if (e instanceof Date) {
+        dateObjects.dateTime = e;
+      }
+      else if (!isNaN(e)) {
+        dateObjects.dateTime = new Date(e);
+      }
+      else if (e === undefined) {
+        dateObjects.dateTime = new Date();
+      }
+      else {
+        console.warn('argument is not instanceof Date');
+      }
+      return dateObjects.dateTime;
+    },
+    getDate: function getDate(e) {
```

Comment:
If nothing is passed to getDate, it should return the date in milliseconds past epoch.